src/functions.jl

```diff
@@ -94,7 +94,13 @@ function Base.permutedims(A::KeyedArray, perm)
     KeyedArray(data, new_keys)#, copy(A.meta))
 end
 
-if VERSION >= v"1.1"
+@static if VERSION > v"1.9-DEV"
+    function Base.eachslice(A::KeyedArray; dims)
+        dims_ix = AxisKeys.dim(A, dims) |> Tuple
+        data = @invoke eachslice(A::AbstractArray; dims=dims_ix)
+        return KeyedArray(NamedDimsArray(data, map(d -> dimnames(A, d), dims_ix)), map(d -> axiskeys(A, d), dims_ix))
+    end
+elseif VERSION >= v"1.1"
     # This copies the implementation from Base, except with numerical_dims:
     @inline function Base.eachslice(A::KeyedArray; dims)
```

Comment:
https://github.com/JuliaLang/Compat.jl/pull/663 still isn't merged, so it makes sense to keep this.